src/GraphQL.MicrosoftDI.Tests/TestExtensions.cs

```diff
@@ -2,9 +2,9 @@ namespace GraphQL.MicrosoftDI.Tests
 {
     internal static class TestExtensions
     {
-        public static void ShouldBeTask(this object value, object expected)
+        public static void ShouldBeTask(this ValueTask<object> value, object expected)
         {
-            value.ShouldBeAssignableTo<Task<object>>().Result.ShouldBe(expected);
+            value.AsTask().Result.ShouldBe(expected);
```

Comment:
```suggestion
            value.Result.ShouldBe(expected);
```

Why not?

Reply:
> The following operations should never be performed on a `ValueTask<TResult>` instance:

> 
> Awaiting the instance multiple times.
> Calling AsTask multiple times.
> Using .Result or .GetAwaiter().GetResult() when the operation hasn't yet completed, or using them multiple times.

See item 3 -- if it hasn't completed, we cannot call Result

Reply:
Per: https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask-1?view=net-6.0

Reply:
Yeah, still need to read all that stuff.